drivers/sensor/lps22df/lps22df_trigger.c

```diff
@@ -0,0 +1,257 @@
+					      GPIO_INT_EDGE_TO_ACTIVE);
+	if (ret < 0) {
+		LOG_ERR("%s: Not able to configure pin_int", dev->name);
+	}
+}
+
+static void lps22df_intr_callback(struct lps22df_data *lps22df)
+{
+#if defined(CONFIG_LPS22DF_TRIGGER_OWN_THREAD)
+	k_sem_give(&lps22df->intr_sem);
+#elif defined(CONFIG_LPS22DF_TRIGGER_GLOBAL_THREAD)
+	k_work_submit(&lps22df->work);
+#endif /* CONFIG_LPS22DF_TRIGGER_OWN_THREAD */
+}
+
+static void lps22df_gpio_callback(const struct device *dev,
+				  struct gpio_callback *cb, uint32_t pins)
+{
+	struct lps22df_data *lps22df =
+		CONTAINER_OF(cb, struct lps22df_data, gpio_cb);
+
+	ARG_UNUSED(pins);
+	const struct lps22df_config *cfg = lps22df->dev->config;
+	int ret;
+
+	ret = gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_DISABLE);
+	if (ret < 0) {
+		LOG_ERR("%s: Not able to configure pin_int", dev->name);
+	}
+
+	lps22df_intr_callback(lps22df);
+}
+
+#ifdef CONFIG_LPS22DF_TRIGGER_OWN_THREAD
+static void lps22df_thread(struct lps22df_data *lps22df)
+{
+	while (1) {
+		k_sem_take(&lps22df->intr_sem, K_FOREVER);
+		lps22df_handle_interrupt(lps22df->dev);
+	}
+}
+#endif /* CONFIG_LPS22DF_TRIGGER_OWN_THREAD */
+
+#ifdef CONFIG_LPS22DF_TRIGGER_GLOBAL_THREAD
+static void lps22df_work_cb(struct k_work *work)
+{
+	struct lps22df_data *lps22df =
+		CONTAINER_OF(work, struct lps22df_data, work);
+
+	lps22df_handle_interrupt(lps22df->dev);
+}
+#endif /* CONFIG_LPS22DF_TRIGGER_GLOBAL_THREAD */
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+static int lps22df_ibi_cb(struct i3c_device_desc *target,
+			  struct i3c_ibi_payload *payload)
+{
+	const struct device *dev = target->dev;
+	struct lps22df_data *lps22df = dev->data;
+
+	ARG_UNUSED(payload);
+
+	lps22df_intr_callback(lps22df);
+
+	return 0;
+}
+#endif
+
+int lps22df_init_interrupt(const struct device *dev)
+{
+	struct lps22df_data *lps22df = dev->data;
+	const struct lps22df_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lps22df_int_mode_t mode;
+	int ret;
+
+	/* setup data ready gpio interrupt */
+	if (!device_is_ready(cfg->gpio_int.port)
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	    && (cfg->i3c.bus == NULL)
+#endif
+	   ) {
+		if (cfg->gpio_int.port) {
+			LOG_ERR("%s: device %s is not ready", dev->name,
+						cfg->gpio_int.port->name);
+			return -ENODEV;
+		}
+
+		LOG_DBG("%s: gpio_int not defined in DT", dev->name);
+		return 0;
+	}
+
+	lps22df->dev = dev;
+
+#if defined(CONFIG_LPS22DF_TRIGGER_OWN_THREAD)
+	k_sem_init(&lps22df->intr_sem, 0, K_SEM_MAX_LIMIT);
+
+	k_thread_create(&lps22df->thread, lps22df->thread_stack,
+		       CONFIG_LPS22DF_THREAD_STACK_SIZE,
+		       (k_thread_entry_t)lps22df_thread, lps22df,
+		       NULL, NULL, K_PRIO_COOP(CONFIG_LPS22DF_THREAD_PRIORITY),
+		       0, K_NO_WAIT);
+#elif defined(CONFIG_LPS22DF_TRIGGER_GLOBAL_THREAD)
+	lps22df->work.handler = lps22df_work_cb;
+#endif /* CONFIG_LPS22DF_TRIGGER_OWN_THREAD */
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	if (cfg->i3c.bus == NULL)
+#endif
```

Comment:
these are quite a bit in the way, how about having a `#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)`  on the top and maybe setting a define as either a condition or true?

```c
#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
        #define ON_I3C_BUS(cfg) (cfg->i3c.bus != NULL)
#else
        #define ON_I3C_BUS false
#endif
```

and then these would be plain `if (!ON_I3C_BUS(cfg)) {`

Reply:
Yes, thanks for suggestion: it will improve a lot the code readibility.
Moreover I saw some other issues on I3C part, so that I will review it all.
Thanks.